lesson15/main.js

```diff
@@ -0,0 +1,98 @@
+"use strict";
+const wrapper = document.getElementById("js-wrapper");
+const ul = document.getElementById("js-ul");
+const submitBtn = document.getElementById("js-submitBtn");
+const modal = document.getElementById("js-modal");
+const openModalBtn = document.getElementById("js-openModalBtn");
+const jsonURL = "https://myjson.dit.upm.es/api/bins/j16z";
+const removeModal = () => {
+   modal.remove();
+   wrapper.classList.remove("bg_gray");
+};
+
+const startLoading = () => {
+   const loadingIcon = document.createElement("img");
+   loadingIcon.id = "loadingIcon";
+   loadingIcon.src = "img/loading-circle.gif";
+   wrapper.appendChild(loadingIcon);
+};
+
+const endLoading = () => {
+   const loadingIcon = document.getElementById("loadingIcon");
+   loadingIcon.remove();
+}
+
+openModalBtn.addEventListener("click", function () {
+   modal.style.display = "block";
+   wrapper.classList.add("bg_gray");
+   openModalBtn.remove();
+});
+
+const isEmptyString = (value) => value === "";
+
+submitBtn.addEventListener("click", () => {
+   const inputNumber = document.getElementById("js-number");
+   const inputText = document.getElementById("js-text");
+   if (isEmptyString(inputNumber.value) || isEmptyString(inputText.value)) { //inputNumber.valueとinputText.valueのデータ型がStringのため
+      alert("未入力です。");
+      return;
+   }
+   console.log(`ID:${inputNumber.value}`);
+   console.log(`ユーザー名:${inputText.value}`);
+   tryCreateElement();
+});
+
+async function init() {
+   removeModal();
+   startLoading();
+   try {
+      return await fetchData();
+   } catch (e) {
+      throw new Error(error);
+   } finally {
+      endLoading();
+   }
+}
+
+async function fetchData() {
+   try {
+      const response = await fetch(jsonURL);
+      const json = await response.json();
+      return json.data;
+   } catch (error) {
+      throw new Error(error);
+   }
+}
+
+async function tryCreateElement() {
+   try {
+      const responseData = await init();
+      if (responseData.length === 0) {
+         const div = document.createElement("div");
+         ul.appendChild(div);
+         div.textContent = `データはまだありません。`;
+         return;
+      }
+      createListElements(responseData);
+   } catch (e) {
+      const div = document.createElement("div");
+      ul.appendChild(div);
+      div.textContent = ` エラーが発生しました：${e}`;
```

Comment:
IMO
This is the same as when responseData is empty, so I think it can be made into a function.

Reply:
Thank you for your review.
I made the function to display a message.
I wanted to separate when the data was empty and when an error occurred, so this is how I did it.
c24020f